src/api.py

```diff
@@ -105,4 +267,10 @@ def run(
                     )
                 )
 
+        file.tags.append(
+            Tag(kind="token_usage",
+                name="token_usage",
+                value=token_usage)
+        )
```

Comment:
I think this tag needs more context.  You could (and we do) run generation on a file multiple times.  Should attribute it with what plugin this is, when it was done, etc